### DISCUSSION TITLE
Why do I get "Error: Unknown command: brew helo" each time Terminal launches?

### DISCUSSION THREAD

**Discussion by Eclipse-2003** [Homebrew/discussions#5538] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.12
ORIGIN: https://github.com/Homebrew/brew
HEAD: 874d2da45344d3b27aa740e555b0210d8c474220
Last commit: 5 days ago
Core tap JSON: 04 Aug 03:07 UTC
Core cask tap JSON: 04 Aug 03:07 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 11
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4/bin/ruby
CPU: 11-core 64-bit arm_lobos
Clang: 15.0.0 build 1500
Git: 2.46.0 => /opt/homebrew/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Hi all,

This is my first time posting on the forums, so I apologize if this post isn't perfect. 
On my Macbook Pro M3 Pro running MacOS Sonoma Version 14.5, I get an error each time I start up Terminal. The issue displays as following:
``` 
Error: Unknown command: brew helo
Example usage:
  brew search TEXT|/REGEX/
  brew info [FORMULA|CASK...]
  brew install FORMULA|CASK...
  brew update
  brew upgrade [FORMULA|CASK...]
  brew uninstall FORMULA|CASK...
  brew list [FORMULA|CASK...]

Troubleshooting:
  brew config
  brew doctor
  brew install --verbose --debug FORMULA|CASK

Contributing:
  brew create URL [--no-fetch]
  brew edit [FORMULA|CASK...]

Further help:
  brew commands
  brew help [COMMAND]
  man brew
  https://docs.brew.sh
```

This is shown immediately after "Last login", I also believe it has always done this ever since I installed it, I wasn't sure if this affected anyone else, which is why I am making a post here rather than a ticket. (I did search for the keyword "helo")
The only time this has changed was recently when I did Brew update and the general troubleshooting before making ticket stuff and it changed the error from "Error: Unknown command: helo" to "Error: Unknown command: brew helo". 

Please let me know if anyone has any ideas, Thanks.

**Comment by gromgit** [comment]
If it only happens _once_ (when you open a new Terminal session), then it's likely `brew helo` is near the end of your shell startup file (`~/.profile` or `~/.bash_profile` for bash; `~/.zprofile` or `~/.zshrc` for zsh).

If it happens _after every command_, then it's in whatever commands you use to set your shell prompt.

(And it's almost certainly a typo of `brew help`, since the `o` and `p` keys are adjacent on QWERTY keyboards.)

**Reply by Eclipse-2003** [reply]
Yep, it only happens once when I open a new Terminal session. I am not sure what you mean by 
> `brew helo` is near the end of your shell startup file (`~/.profile` or `~/.bash_profile` for bash`;` /.zprofile`or`/.zshrc` for zsh).

And I'm not sure how to go about fixing it. My Terminal says zsh at the top but I am confused. 
I am also confused when you mention it's a typo of `brew help` Just to be clear this happens automatically after opening Terminal with no keystrokes.

**Reply by gromgit** [reply]
Since your Terminal says "zsh", what's the output of the following command:
```
zsh -xli </dev/null 2>&1 | tee /tmp/trace.log | grep "brew helo"
```
> I am also confused when you mention it's a typo of `brew help`

If someone wanted to run `brew help`, they could easily type `brew helo` instead, since the `o` and `p` keys are right next to each other on US/UK-style (QWERTY) keyboards. It's also _not_ something that would've been added by a script, since scripts don't make such mistakes (though script _writers_ might).

**Reply by Eclipse-2003** [reply]
This is the output i get after running that command:
```
+/Users/[username]/.zprofile:5> brew helo
Error: Unknown command: brew helo
```

And that makes sense since they are easy to mistype but my confusion mainly stems from why it is automatically entering that each Terminal session.

**Reply by gromgit** [reply]
OK, so the command output says the problem is line 5 in your `~/.zprofile`, so you should open the file in a text editor and remove that line, e.g.:
```
open -a TextEdit ~/.zprofile
```

> why it is automatically entering that each Terminal session.

Because your Terminal is starting `zsh` as a login shell for each window, which makes `zsh` run the contents of `~/.zprofile`.  `man zsh` and read the **STARTUP/SHUTDOWN FILES** section for details.

**Reply by Eclipse-2003** [reply]
So I went into the file with the command you posted and deleted that line, and now when I boot terminal I get this:
```
Last login: Sat Aug  3 22:07:48 on ttys000
Example usage:
  brew search TEXT|/REGEX/
  brew info [FORMULA|CASK...]
  brew install FORMULA|CASK...
  brew update
  brew upgrade [FORMULA|CASK...]
  brew uninstall FORMULA|CASK...
  brew list [FORMULA|CASK...]

Troubleshooting:
  brew config
  brew doctor
  brew install --verbose --debug FORMULA|CASK

Contributing:
  brew create URL [--no-fetch]
  brew edit [FORMULA|CASK...]

Further help:
  brew commands
  brew help [COMMAND]
  man brew
  https://docs.brew.sh
```

Super strange behavior, here is the contents of that document:
```

eval "$(/opt/homebrew/bin/brew shellenv)"
 
 

brew help
```

As a note, just above brew help is where `brew helo` was, I'm assuming now it's reading the blank space as a command but idk, I'll try running the update stuff again along with doctor.

**Reply by carlocab** [reply, marked as answer]
Remove `brew help` from that file and the output when you open the terminal should go away.

**Reply by Eclipse-2003** [reply]
Hi, I think that worked, seems like it was just reading those commands on startup, Thanks a ton!
